Add Using Webhooks to PEM documentation (#5258)
EnterpriseDB/docs: product_docs/docs/pem/9/monitoring_performance/notifications.mdx

-   To send a notification when a specific alert level is reached, set the slider next to an alert level to **Yes**. You must set **All alerts** to **No** to configure an individual alert level.

### Example of configuring a webhook using Slack

Slack provides a way to post messages via `Incoming Webhooks`. To send an alert from PEM to Slack, users must create an `Incoming Webook`. Refer to Slack's guide [Sending messages using incoming webhooks](https://api.slack.com/messaging/webhooks). The `Incoming Webhook` gives a unique URL to which the pemAgent can send a JSON payload with the message text and other options to Slack webhooks.

Here are the steps to configure a Slack `Incoming Webhook` using PEM:

- Configure the Slack application, if it's not already configured.
- Activate `Incoming Webhooks` from Slack.
- Add new webhooks to the Slack workspace. You must add the channel and authorize the application for the incoming webhooks so the pemAgent can notify the alert using Slack.
- Get the unique URL used in PEM to configure the webhook configurations. You can configure PEM to send notifications (low, medium, high) to Slack via a POST request.

> - The pemAgent uses the URL to send alert notifications from PEM to Slack using an HTTP POST request, as configured in the webhook.  

Reviewer: I think this line is trying to say that the user should enter the unique URL generated by Slack into the Webhook configuration in PEM (as described above), if so we need to correct the wording. In any case this whole section needs much more explicit instructions.

> Here is an example of a POST request from the pemAgent:

Reviewer: ```suggestion
Here is an example of a POST request from the PEM Agent:
```

-   Select **Test Connection** to test notification delivery to the mentioned endpoint.

### Defining webhook alert levels

Use the **Notifications** tab to specify an alert level for webhook endpoints:

-   Set **All alerts** to **Yes** to enable all alert levels to send notifications.
-   To send a notification when a specific alert level is reached, set the slider next to an alert level to **Yes**. You must set **All alerts** to **No** to configure an individual alert level.

### Example: sending notifications to Slack

In Slack, follow the instructions in [Getting started with incoming webhooks](https://api.slack.com/messaging/webhooks) to:

> - Create a Slack app

Reviewer 2: ```suggestion
- Download an login to a Slack app
```

Reviewer 3: Creating a slack app is correct. Slack apps live in Slack server side and are created on the web UI, there is nothing to download.

Reviewer: Existing wording is correct as pointed out by DJ.

- Activate incoming webhooks for that app
- Add a webhook that posts to a channel or user of your choice

> The newly created webhook will have a unique URL something like https://hooks.slack.com/services/x/y/z we can now configure PEM to send notifications to this URL.

Reviewer 2: ```suggestion
The newly created webhook will have a unique URL something like https://hooks.slack.com/services/x/y/z . We can now configure PEM to send notifications to this URL.
```

In PEM, [create a new webhook](#creating-a-webhook), give it a descriptive name and copy the URL you obtained above to the 'URL' field.
Ensure that 'Request method' is set to 'POST' and 'Enable?' is set to 'Yes'. Set all the sliders under 'Alert Notifications' to 'Yes'.

> Under HTTP headers, add a header with the key `Content-Type` and the value `application/json`.

Reviewer 2: ```suggestion
Add a header under HTTP headers with the key `Content-Type` and the value `application/json`.
```

Under Payload, delete the default template and specify a template with `text` as the top-level key as in the example below.

```
{"text": "%AlertName% on %ObjectType% %ObjectName% is now %CurrentState%"}
```

> You can now test the connection. If it succeeds, you will get a notification in PEM and the template above will appear in your Slack channel as a message.

Reviewer 2: ```suggestion
You can now test the connection. If it succeeds, you will get a notification in PEM, and the template above will appear in your Slack channel as a message.
```

> Save the webhook and continue using PEM as normal. All alerts will now be sent to your Slack channel!

Reviewer 2: ```suggestion
Save the webhook and continue using PEM as usual. Now, PEM will send all the alerts to your Slack channel.
```

-   Set **All alerts** to **Yes** to enable all alert levels to send notifications.
-   To send a notification when a specific alert level is reached, set the slider next to an alert level to **Yes**. You must set **All alerts** to **No** to configure an individual alert level.

### Using a webhook

> Slack provides a way to post messages via `Incoming Webhooks`. To send an alert from PEM to Slack, users must create an `Incoming Webook`. Refer to Slack's guide [Sending messages using incoming webhooks`](https://api.slack.com/messaging/webhooks). The `Incoming Webhook` gives a unique URL to which the pemAgent can send a JSON payload with the message text and other options to Slack webhooks.

Reviewer 2: ```suggestion
Slack provides a way to post messages via `Incoming Webhooks`. To send an alert from PEM to Slack, users must create an `Incoming Webook`. Refer to Slack's guide [Sending messages using incoming webhooks](https://api.slack.com/messaging/webhooks). The `Incoming Webhook` gives a unique URL to which the pemAgent can send a JSON payload with the message text and other options to Slack webhooks.
```

-   Set **All alerts** to **Yes** to enable all alert levels to send notifications.
-   To send a notification when a specific alert level is reached, set the slider next to an alert level to **Yes**. You must set **All alerts** to **No** to configure an individual alert level.

### Using a webhook

> Slack provides a way to post messages via `Incoming Webhooks`. To send an alert from PEM to Slack, users must create an `Incoming Webook`. The `Incoming Webhook` gives a unique URL to which the pemAgent can send a JSON payload with the message text and other options to Slack webhooks.

Reviewer 3: ```suggestion
Slack provides a way to post messages via `Incoming Webhooks`. To send an alert from PEM to Slack, users must create an `Incoming Webook`. Refer to Slack's guide [Sending messages using incoming webhooks`](https://api.slack.com/messaging/webhooks). The `Incoming Webhook` gives a unique URL to which the pemAgent can send a JSON payload with the message text and other options to Slack webhooks.
```

Here are the steps to configure a Slack `Incoming Webhook` using PEM:

- Configure the Slack application, if it's not already configured.
- Activate `Incoming Webhooks` from Slack.
- Add new webhooks to the Slack workspace. You must add the channel and authorize the application for the incoming webhooks so the pemAgent can notify the alert using Slack.
- Get the unique URL used in PEM to configure the webhook configurations. You can configure PEM to send notifications (low, medium, high) to Slack via a POST request.
- The pemAgent uses the URL to send alert notifications from PEM to Slack using an HTTP POST request, as configured in the webhook.  

Here is an example of a POST request from the pemAgent:

```shell
POST https://hooks.slack.com/services/T00000000/B00000000/XXXXXXXXXX/
Content-type: application/json
{
    "text": "Alert from PEM: [%AlertID%], [%AlertName%]"
}
```

!!! Note

>     To configure Slack, see 

Reviewer 3: I assume this was going to take the link to the slack docs?

Author: Removed stub because DJH added a.link earlier
